Automattic/jetpack
# Gif block: refactor and add test coverage

#### Changes proposed in this Pull Request:


After trying to add test coverage to the Gif block in https://github.com/Automattic/jetpack/pull/19019, we thought we'd preserve some sanity by refactoring the block in order to:

- Make the code more readable. 
- Make it all more testable. 
- Make life more comfortable

#### Jetpack product discussion
p1HpG7-b3G-p2

#### Does this pull request change what data or activity we track or use?
Nope.

#### Testing instructions:

Make sure there are no visual or functional regressions in the editor or the save versions. Nothing should change in this respect.

Check that:

- [ ] The search function works as expected, also try the accepted giphy URL formats, e.g., https://media.giphy.com/media/L0B6dlBlmxo6dt6YkI/giphy.gif, https://giphy.com/gifs/nfl-football-super-bowl-lv-kxrANnpcIlNiGPViUn, http://i.giphy.com/hvE0PhVAnGQAo.gif and  https://giphy.com/embed/35Co9iXhs8sHgmCZFc
- [ ] You can select a giphy to display from the search results' thumbnails
- [ ] You can write and save a caption

![Mar-09-2021 22-07-25](https://user-images.githubusercontent.com/6458278/110461758-f178db00-8123-11eb-92b7-46a3027c11f4.gif)

<img width="850" alt="Screen Shot 2021-03-09 at 10 06 22 pm" src="https://user-images.githubusercontent.com/6458278/110461712-e625af80-8123-11eb-949c-d5dab52cf342.png">

Run the unit tests:

```
cd projects/plugins/jetpack
yarn fixtures:test extensions/blocks/gif/test
```

#### Proposed changelog entry for your changes:

Not sure if a changelog is required for a refactor and tests. No functionality/UI has changed.

* GIF Block: refactored edit.js and added test coverage
